src/screens/PrivateChat/ContactsPicker/index.js

```diff
@@ -0,0 +1,259 @@
+// @flow
+
+import React from 'react';
+import { connect } from 'react-redux';
+import {
+  View,
+  ScrollView,
+  TouchableOpacity,
+  Clipboard,
+  Image,
+} from 'react-native';
+import { Text } from 'native-base';
+import { getProfile, newChatSession } from '../../../actions/chat';
+import { addContact } from '../../../actions/contacts';
+import BackgroundImage from '../../../components/common/BackgroundImage';
+import styles from './styles';
+import FakeNavigationBar from '../../../components/common/FakeNavigationBar';
+import Loading from '../../../components/common/Loading';
+import NavigatorComponent from '../../../components/common/NavigatorComponent';
+import i18n from '../../../global/i18n';
+import Colors from '../../../global/colors';
+import type { ProfileType } from '../../../types/Chat';
+import ScreenTitle from '../../../components/common/ScreenTitle';
+import InvalidKeyModal from './InvalidKeyModal';
+import AssetsImage from '../../../global/AssetsImages';
+import { imageSource } from '../../../utils/profile';
+
+import { Chip, Selectize } from 'react-native-material-selectize';
+
+const DONE_BUTTON = 'DONE_BUTTON';
+const INVALID_MODAL_KEY = 'invalidKey';
+
+type Props = {
+  /**
+   * @desc React Native Navigation navigator object.
+   */
+  navigator: Navigator,
+  /**
+   * @desc List of all contacts
+   */
+  contacts: Array<*>,
+  /**
+   * @desc Function to get user profile
+   * @param {string} identityKey Identity key of user.
+   * @param {function} callback Callback
+   */
+  getProfile: (identityKey: string, callback: (profile: (ProfileType | null), error: (Error | null)) => void) => void,
+  /**
+   * @desc Function to add a new contact
+   * @param {string} identityKey Identity key of user.
+   * @param {function} callback Callback
+   */
+  addContact: (identityKey: string, callback: (error: Error | null) => void) => void,
+  /**
+   * @desc Function to initialize a new chat
+   * @param {ProfileType} profile Profile of the user
+   * @param {func} callback
+   */
+  createNewSession: (profile: ProfileType, callback: (result: Object) => void) => void,
```

Comment:
Picker shouldn't care about chat session creating, it's just a component to select set of contacts. Because it can be used to e.g. modifying list of members of existing chat. So, picker should just return array of selected contacts via callback like `onContactsSelected`.